### DISCUSSION TITLE
Add contributors file for zebra

### DISCUSSION THREAD

**Discussion by oxarbitrage** [ZcashFoundation/zebra#3637] [Ideas]
We want to do some research on adding contributors file for zebra but we will prefer to be an automatic process so we don't increase the amount of work needed to create releases.

This ticket is to keep track of this research, more details will be added.

**Comment by oxarbitrage** [comment]
I had been doing some research on this and trying different options, here is more or less what i got:

There are a few options to use github actions to automate the process however some of the ones i tried didn't worked for me and the ones that did ((not without some issues) are not convincing.

The most developed option seems to be the one that @mpguerra recommended me: https://allcontributors.org/. This is a bot but also a command line tool. I realized that a good setup can be to submit to zebra an initial table of contributors created locally by the the command line tool and then let the bot handle new contributors. The good thing about the bot is that contributors can be be added by the contributor themselves but it creates one PR per addition, that is a bit annoying for the first batch so i will prefer to create locally and then import it to zebra.

So i created a script to do the initial list, please feel free to let me know if you want to change your contributor icons!

https://github.com/oxarbitrage/test-all-contributors-bot/

- The [README](https://github.com/oxarbitrage/test-all-contributors-bot/blob/main/README.md) has the commands used to create the list.

- [CONTRIBUTORS](https://github.com/oxarbitrage/test-all-contributors-bot/blob/main/CONTRIBUTORS.md) has the generated table sorted alphabetically.

- [CONTRIBUTING](https://github.com/oxarbitrage/test-all-contributors-bot/blob/main/CONTRIBUTING.md) has the text addition we need to do to the zebra `CONTRIBUTING.md` file.

- [.all-contributorsrc](https://github.com/oxarbitrage/test-all-contributors-bot/blob/main/.all-contributorsrc) is the all-contributors config file that both the bot and the command line tool can recognize.

I really don't know if this is out best option, we can discuss about it.

**Comment by mpguerra** [comment]
moving this to our unofficial documentation sprint (Sprint 24) towards the end of the year. I think this will be a nice addition when we are getting ready to have zebra more widely available.

Also, some of the automation can wait for the devops engineer.

**Comment by teor2345** [comment]
We could just link to the GitHub contributors view for now.

It's not perfect, because it doesn't detect PRs submitted for someone else. But it's a quick way to start it off.

**Comment by dconnolly** [comment]
Not strictly required for release candidate // @mpguerra @ftm1000

**Comment by oxarbitrage** [comment]
I don't think doing this brings too much value to the project at this stage and the discussion had been open for a while without moving. Closing.
